lua/orgmode/objects/date.lua

```diff
@@ -11,25 +11,31 @@ local time_format = '%H:%M'
 
 ---@alias OrgDateSpan 'minute' | 'hour' | 'day' | 'week' | 'month' | 'year'
 
----@class OrgDate
----@field type string
+---@class OsTime
+
+---@class OsDate
+---@field day number?
```

Comment:
I think all of the fields should be required, at least `os.date('*t', timestamp)` always returns all of these.
Unless you did it because of some existing logic that is not handling it, which we can discuss.

Reply:
Actually I made the field optional to fix the warnings when assigning the result of the `tonumber` functions in `parse_datetime`. The other option would be to provide a default, if tonumber returns nil.

https://github.com/nvim-orgmode/orgmode/blob/ac9a6e0eac2d298a27ab4f59511e60897cb61a08/lua/orgmode/objects/date.lua#L150-L174

I mean, you have a point - because OrgDate transitively inherits from OsDate (as it is currently implemented) it inherits also the optionals, which we might not want to. Should I fill in some sane defaults in the parsing function and if so, which one should we choose? We could just take the current date or zero. For the year, I think the current year makes more sense, same might be the case for month and even day. What do you think?

Reply:
You can default it to `0`, because at this point the value should already be validated and we should have all the numbers.

Reply:
Done.